modules/packages/LinearAlgebra.chpl

```diff
@@ -1886,6 +1886,35 @@ module Sparse {
   /* Transpose CSR matrix */
   proc _array.T where isCSArr(this) { return transpose(this); }
 
+  /* Transpose Sparse domain */
+  proc transpose(D: domain) where isSparseDom(D) && !isCSDom(D) {
+    var indices: [1..0] 2*D.idxType;
+    for (i,j) in D {
+      indices.push_back((j,i));
+    }
+    
+    const parentDT = transpose(D.parentDom);
+    var Dom: sparse subdomain(parentDT);
+    Dom += indices;
+    return Dom;
+  }
+
+  /* Transpose Sparse matrix */
+  proc transpose(A: [?Adom] ?eltType) where isSparseArr(A) && !isCSArr(A) {
+    var Dom = transpose(Adom);
+    var B: [Dom] eltType;
+  
+    forall (i,j) in Adom {
+      B[j, i] = A[i, j];
+    }
+    return B;
+  }
+
+  /* Transpose Sparse matrix */
+  proc _array.T where isSparseArr(this) && !isCSArr(this) { 
```

Comment:
This is fine as is, but I wonder if we could combine the 2 `_array.T` wrappers into:

```chpl
proc _array.T where isSparseArr(this) { return transpose(this); }
```

where `transpose(this)` will resolve to the correct sparse type. (I'm also wondering if we could do this across all arrays regardless of sparse/dense, but let's not tackle that here.)